cpp/src/community/ktruss_impl.cuh

```diff
@@ -0,0 +1,892 @@
+    printf("\nthe vertex pair buffer size = %d\n", vertex_pair_buffer_size);
+    //auto x = thrust::get<1>(*invalid_edge_first);
+    //printf("\nx = %d\n", x);
+    // Run binary search
+    //if (i < vertex_pair_buffer_size) {
+    if (i < num_invalid_edges) {
+      auto pair = thrust::get<0>(*(invalid_edge_first + i)); // FIXME only increment if it is less than 'invalid_edge_first.size()'
+      printf("\nsrc_pair = %d, dst_pair = %d\n", thrust::get<0>(pair), thrust::get<1>(pair));
+    
+      auto itr = thrust::upper_bound(thrust::seq,
+                      vertex_pair_buffer,
+                      vertex_pair_buffer + vertex_pair_buffer_size,
+                      //pair,
+                      thrust::make_tuple(thrust::get<0>(pair), thrust::get<1>(pair)),
+                      thrust::less<thrust::tuple<vertex_t, vertex_t>>{});
+      
+      printf("\n itr = %d\n", itr);
+      auto idx = thrust::distance(vertex_pair_buffer + 1, itr);
+      printf("\n idx = %d \n", idx);
+      
+      return 1;
+    }
+    else {
+      return 0;
+    }
+
+
+   return 0;
+  }
+};
+
+
+
+
+
+
+
+
+
+template <typename vertex_t, typename edge_t>
+struct intersection_op_t {
+  __device__ thrust::tuple<edge_t, edge_t> operator()(
+    vertex_t v0,
+    vertex_t v1,
+    edge_t v0_prop /* out degree */,
+    edge_t v1_prop /* out degree */,
+    raft::device_span<vertex_t const> intersection,
+    std::byte, /* dummy */
+    std::byte  /* dummy */
+  ) const
+  {
+    return thrust::make_tuple(v0_prop + v1_prop, static_cast<edge_t>(intersection.size()));
+  }
+};
+
+
+} // namespace
+
+template <typename vertex_t, typename edge_t, bool multi_gpu>
+void ktruss(raft::handle_t const& handle,
+            graph_view_t<vertex_t, edge_t, false, multi_gpu> const& graph_view,
+            edge_t k,
+            bool do_expensive_check)
+{
+  using weight_t = float;  // dummy
+
+
+  // 1. Check input arguments.
+
+  CUGRAPH_EXPECTS(!graph_view.has_edge_mask(), "unimplemented.");
+
+  CUGRAPH_EXPECTS(
+    graph_view.is_symmetric(),
+    "Invalid input arguments: triangle_count currently supports undirected graphs only.");
+  CUGRAPH_EXPECTS(
+    !graph_view.is_multigraph(),
+    "Invalid input arguments: triangle_count currently does not support multi-graphs.");
+  
+
+  // 2. Exclude self-loops (FIXME: better mask-out once we add masking support).
+
+  std::optional<graph_t<vertex_t, edge_t, false, multi_gpu>> modified_graph{std::nullopt};
+  std::optional<graph_view_t<vertex_t, edge_t, false, multi_gpu>> modified_graph_view{std::nullopt};
+  std::optional<rmm::device_uvector<vertex_t>> renumber_map{std::nullopt};
+  std::optional<edge_property_t<graph_view_t<vertex_t, edge_t, false, multi_gpu>, edge_t>> edge_ids{std::nullopt};
+  // FIXME: Maybe should not be optional
+  std::optional<rmm::device_uvector<edge_t>> wgts{std::nullopt};
+  std::optional<rmm::device_uvector<edge_t>> wgts_{std::nullopt};
```

Comment:
If you recalled I was initially trying to store the number of triangles in edge_ids but now I am updating the edge property with `transform_e`. I just didn't clean up my code since that but those variables are removed now